lua/entities/gmod_wire_expression2/core/custom/prop.lua

```diff
@@ -50,43 +50,77 @@ local function MakePropNoEffect(...)
-function PropCore.CreateProp(self,model,pos,angles,freeze)
-	if(!util.IsValidModel(model) || !util.IsValidProp(model) || not PropCore.ValidSpawn() )then
-		return nil
+function PropCore.CreateProp(self,model,pos,angles,freeze,isVehicle)
+
+	if not PropCore.ValidSpawn() then return nil end
+	
+	if isVehicle then
+		if model == "" then model = "models/nova/airboat_seat.mdl" end
+		if model ~= "models/vehicles/prisoner_pod_inner.mdl" then angles = angles + Angle(0, -90, 0) end
 	end
+	
+	if not util.IsValidModel( model ) or not util.IsValidProp( model ) then return nil end
+	
 	pos = E2Lib.clampPos( pos )
+	
 	local prop
 	
-	if self.data.propSpawnEffect then
-		prop = MakeProp( self.player, pos, angles, model, {}, {} )
+	local cleanupCategory = "props"
+	local undoCategory = "e2_spawned_prop"
+	local undoName = "E2 Prop"
+	
+	if isVehicle then
+		cleanupCategory = "vehicles"
+		undoCategory = "e2_spawned_seat"
+		undoName = "E2 Seat"
+		
+		prop = ents.Create("prop_vehicle_prisoner_pod")
+		prop:SetModel(model)
+		prop:SetPos(pos)
+		prop:SetAngles(angles)
+		
+		if self.data.propSpawnEffect then DoPropSpawnedEffect( prop ) end
+		
+		prop:Spawn()
+		prop:SetKeyValue( "limitview", 0 )
+		
+		table.Merge( prop, { HandleAnimation = function( _, ply ) return ply:SelectWeightedSequence( ACT_HL2MP_SIT ) end } )
+		gamemode.Call( "PlayerSpawnedVehicle", self.player, prop )
 	else
-		prop = MakePropNoEffect( self.player, pos, angles, model, {}, {} )
+		prop = self.data.propSpawnEffect and MakeProp( self.player, pos, angles, model, {}, {} ) or MakePropNoEffect( self.player, pos, angles, model, {}, {} )
 	end
+	
 	if not prop then return end
 	
 	prop:Activate()
-	self.player:AddCleanup( "props", prop )
-	if self.data.propSpawnPersist then
-		undo.Create("e2_spawned_prop")
+	
+	self.player:AddCleanup( cleanupCategory, prop )
+	
+	if self.data.propSpawnUndo then
+		undo.Create( undoCategory )
 			undo.AddEntity( prop )
 			undo.SetPlayer( self.player )
-		undo.Finish()
-	else
-		table.insert( self.data.propSpawns, prop )
+		undo.Finish( undoName .. " (" .. model .. ")" )
 	end
-	table.insert( self.data.propSpawnsAll, prop )
-	
+
 	local phys = prop:GetPhysicsObject()
-	if (phys:IsValid()) then
-		if(angles!=nil) then E2Lib.setAng( phys, angles ) end
+	if IsValid( phys ) then
+		if angles ~= nil then E2Lib.setAng( phys, angles ) end
 		phys:Wake()
-		if(freeze>0) then phys:EnableMotion( false ) end
+		if freeze > 0 then phys:EnableMotion( false ) end
 	end
-	prop:CallOnRemove( "wire_expression2_propcore_remove", function( prop )
-		E2totalspawnedprops = E2totalspawnedprops - 1
-	end)
-	E2totalspawnedprops = E2totalspawnedprops+1
-	E2tempSpawnedProps = E2tempSpawnedProps+1
+	
+	prop:CallOnRemove( "wire_expression2_propcore_remove",
+		function( prop )
+			self.data.spawnedProps[ tostring( prop ) ] = nil
+			E2totalspawnedprops = E2totalspawnedprops - 1
+		end
+	)
+	
+	self.data.spawnedProps[ tostring( prop ) ] = { ent = prop, undo = self.data.propSpawnUndo }
```

Comment:
Wire usually uses entities themselves as the keys in tables, so self.data.spawnedProps[prop] instead of self.data.spawnedProps[ tostring( prop ) ]